File: src/main/java/de/maggiwuerze/xdccloader/service/ServerService.java
package de.maggiwuerze.xdccloader.service;

import de.maggiwuerze.xdccloader.model.entity.Server;
import de.maggiwuerze.xdccloader.persistence.ServerRepository;
import java.util.List;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
@Slf4j
public class ServerService {

	private final ServerRepository serverRepository;

	public List<Server> list() {
		return serverRepository.findAll();
	}

	public Server save(Server server) {
		return serverRepository.save(server);
	}

	public Server findById(Long serverId) {
		return serverRepository.findById(serverId).orElse(null);
	}

	public void delete(Long serverId) {serverRepository.deleteById(serverId);}
}

File: src/main/kotlin/de/maggiwuerze/xdccwebloader/service/ServerService.kt
package de.maggiwuerze.xdccwebloader.service

import de.maggiwuerze.xdccwebloader.model.entity.Server
import de.maggiwuerze.xdccwebloader.model.forms.ServerFormTO
import de.maggiwuerze.xdccwebloader.persistence.ServerRepository
import org.springframework.stereotype.Service
import java.util.*
import kotlin.jvm.optionals.getOrNull

@Service
class ServerService(private val serverRepository: ServerRepository) {

    fun list(): List<Server> {
        return serverRepository.findAll()
    }

    fun save(server: Server): Server {
        return serverRepository.save(server)
    }

    fun findById(serverId: UUID): Server? {
        return serverRepository.findById(serverId).orElse(null)
    }

    fun delete(serverId: UUID) {
        serverRepository.deleteById(serverId)
    }

    fun update(id: UUID, serverFormTO: ServerFormTO): Server {
        return serverRepository.findById(id).getOrNull()?.let {
            it.name = serverFormTO.name
            it.serverUrl = serverFormTO.serverUrl
            serverRepository.save(it)
        } ?: throw IllegalStateException("Server with id $id not found.")
    }
}
